src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
-            throw new TypeError(`Expected instance of RunwayModel, but received ${runwayModel}`);
+    _initializePositionInRouteToBeginAtFixName(fixName, category) {
+        if (category === FLIGHT_CATEGORY.DEPARTURE) {
+            return;
         }
 
-        if (this.departureRunwayModel && this.departureRunwayModel.name === runwayModel.name) {
-            return;
+        if (_isNil(fixName)) {
+            return this.moveToNextWaypoint();
         }
 
-        this.departureRunwayModel = runwayModel;
+        if (!this._routeModel.hasWaypointName(fixName)) {
+            throw new TypeError(`Expected initial fix to be in flight plan route, but received '${fixName}'`);
+        }
 
-        this._regenerateSidLeg();
+        this.skipToWaypointName(fixName);
     }
 
-    /**
-     * Encapsulates setting of `#arrivalRunwayModel`
-     *
-     * @for fms
-     * @method setArrivalRunway
-     * @param runwayModel {RunwayModel}
-     */
-    setArrivalRunway(runwayModel) {
-        // TODO: this should be an `instanceof` check and should be implemented as part of (or after)
-        // https://github.com/openscope/openscope/issues/93
-        if (!_isObject(runwayModel)) {
-            throw new TypeError(`Expected instance of RunwayModel, but received ${runwayModel}`);
-        }
+    // ------------------------------ PUBLIC ------------------------------
 
-        if (this.arrivalRunwayModel && this.arrivalRunwayModel.name === runwayModel.name) {
-            return;
+    activateHoldForWaypointName(waypointName, holdParameters) {
+        if (!this._routeModel.hasWaypointName(waypointName)) {
+            return [false, `unable to hold at ${waypointName}; it is not on our route!`];
         }
 
-        this.arrivalRunwayModel = runwayModel;
-
-        this._regenerateStarLeg();
+        this._routeModel.activateHoldForWaypointName(waypointName, holdParameters);
     }
 
     /**
-     * Set the `#currentPhase`
-     *
-     * this value is used to determine how to calculate and aircraft's next
-     * altitude, heading and speed.
+     * Return an array of waypoints in the flight plan that have altitude restrictions
      *
      * @for Fms
-     * @method setFlightPhase
-     * @param phase {string}
+     * @method getAltitudeRestrictedWaypoints
+     * @return {array<WaypointModel>}
      */
-    setFlightPhase(phase) {
-        if (!_has(FLIGHT_PHASE, phase)) {
-            return new TypeError(`Expected known flight phase, but received '${phase}'`);
-        }
+    getAltitudeRestrictedWaypoints() {
+        return this._routeModel.getAltitudeRestrictedWaypoints();
+    }
 
-        if (this.currentPhase === phase) {
-            return;
-        }
+    /**
+    * Returns the lowest bottom altitude of any `LegModel` in the `#_legCollection`
+    *
+    * @for Fms
+    * @method getBottomAltitude
+    * @return {number}
+    */
+    getBottomAltitude() {
+        return this._routeModel.getBottomAltitude();
+    }
+
+    /**
```

Comment:
fix spacing.  there are a couple of these in this class, take a spin through and see what you find.